wgsl/index.bs

```diff
@@ -8185,14 +8185,75 @@ Note: The [=attribute/const=] attribute cannot be applied to user-declared funct
     * An [[#address-of-expr|address-of expression]] of a
         [[#var-identifier-expr|variable identifier expression]]
     * A [=formal parameter|function parameter=]
+    * A [[#constant-identifier-expr|constant identifier expression]] or
+        [[#address-of-expr|address-of expression]] where the only expressions
+        applied to the [=root identifier=] would satisfy the previous two
+        rules.
+        That is, the [=memory view=] of the [=root identifier=] is unchanged.
+        This means no [[#vector-access-expr|vector]],
+        [[#matrix-access-expr|matrix]], [[#array-access-expr|array]], or
+        [[#struct-access-expr|struct]] access expressions can be applied to
+        produce a [=memory view=] into the root identifier when traced
+        from the argument back through all the [=let-declarations=].
 
 Note: Recursion is disallowed because cycles are not permitted among any kinds
 of declarations.
 
+<div class='example wgsl' heading='Valid and invalid pointer arguments'>
+  <xmp highlight='rust'>
+    fn bar(p : ptr<function, f32>) {
+    }
+
+    fn baz(p : ptr<private, i32>) {
+    }
+
+    fn bar2(p : ptr<function, f32>) {
+      let a = &*&*&(p);
+
+      bar(p); // Valid
+      bar(a); // Valid
+    }
+
+    struct S {
+      x : i32
+    }
+
+    var usable_priv : i32;
+    var unusable_priv : array<i32, 4>;
+    fn foo() {
+      var usable_func : f32;
+      var unuable_func : S;
```

Comment:
`ususable_func`?

Although annotating these variables with `usable` and `usable` is a little confusing, because it's not the variable type that makes them usable or not - it's the expressions that wrap these.

Reply:
Well there is no way given the function declarations that take pointers to make those variable usable, that's why I chose that naming.

Reply:
You could pass a direct pointer to `ususable_func` as a function parameter, no?

Reply:
That is in the examples below.

Reply:
Oh I see what you mean, yes, but the context is related to `bar`, `bar2`, and `baz` declarations.